sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala

```diff
@@ -629,7 +627,12 @@ class RocksDBSuite extends AlsoTestWithRocksDBFeatures with SharedSparkSession
 
       if (isChangelogCheckpointingEnabled) {
         assert(changelogVersionsPresent(remoteDir) === (1 to 50))
-        assert(snapshotVersionsPresent(remoteDir) === Range.inclusive(5, 50, 5))
+        if (colFamiliesEnabled) {
+          assert(snapshotVersionsPresent(remoteDir) ===
+            Seq(1, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50))
```

Comment:
Wait, we should have set the flag on enforcing snapshot for change of column family before this fix. Does this mean we had a bug?

Reply:
No - this just additionally sets the flag for the first batch as well. so just added 1 essentially